api/actions/loadArticles.js

```diff
@@ -14,8 +15,53 @@ export default function loadArticles(req, params = []) {
       if (err) {
         reject(err)
       } else {
-        resolve(res.body)
+        const queryObj = JSON.parse(_.get(query, 'embedded'))
+        let aricleRes = res.body
+        let writers =  []
+        const list = [ 'writters', 'photographers', 'designers', 'engineers' ]
+        list.forEach((item) => {
+          let aArr = _.get(aricleRes, item)
+          aArr.forEach((author) => {
+            if(author.image) {
+              writers.push('"' + author.image + '"')
+            }
+          })
+        })
+
+        // combine author images data if the query contains 'authorImages'
+        if(_.get(queryObj, 'authorImages')) {
+          const imgIds = _.uniq(writers)
+          const imgQuery = encodeURI(`where={"_id":{"$in":[${imgIds}]}}`)
+          const imgUrl = `${API_PROTOCOL}://${API_HOST}:${API_PORT}/images?${imgQuery}`
+
+          superAgent['get'](imgUrl).timeout(500)
+          .end(function (err, res) {
+            if (err) {
+              reject(err)
```

Comment:
we should not reject it here, because we still can render the page without authors images.